paddleslim/prune/collections.py

```diff
@@ -0,0 +1,226 @@
+"""Define some functions to collect ralated parameters into groups."""
+# Copyright (c) 2020  PaddlePaddle Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import copy
+import logging
+from ..core import GraphWrapper, VarWrapper
+from ..common import get_logger
+from .prune_worker import PRUNE_WORKER, UnsupportOpError
+
+__all__ = [
+    'PruningDetails', 'PruningCollection', 'PruningCollections',
+    'StaticPruningCollections'
+]
+
+_logger = get_logger(__name__, level=logging.INFO)
+
+
+class PruningDetails(object):
+    """
+    The description of one pruning operation.
+    Args:
+        var(VarWrapper): The variable to be pruned.
+        axis(int): The axis to be pruned on.
+        transform(dict): Information used to convert pruned indexes of master
+                         tensor to indexes of current tensor.
+        op(OpWrapper): The operator with current tensor as input.
+        is_parameter(bool): whether the tensor is parameter. Default: True.
+    """
+
+    def __init__(self, var, axis, transform, op, is_parameter=True):
+        assert (isinstance(var, VarWrapper),
+                "name should be VarWrapper, but get type = ".format(type(var)))
+        assert (isinstance(axis, int))
+        self.name = var.name()
+        self.var = var
+        self.axis = axis
+        self.transform = transform
+        self.op = op
+        self.is_parameter = is_parameter
+
+    def __eq__(self, other):
+        if self.name != other.name:
+            return False
+        if self.axis != other.axis:
+            return False
+        if self.transform != other.transform:
+            return False
+        return True
+
+
+class PruningCollection(object):
+    """
+    A group of pruning operations.
+    
+      conv1-->conv2-->batch_norm
+
+    For the network defined above, if weight of conv1 is pruned on 0-axis,
+    weight of'conv2' should be pruned on 1-axis. The pruning operations on 0-axis of
+    'conv1' and that on 1-aixs of 'conv2' is a collection. And the {'name': conv1.weight_name, 'axis': 0}
```

Comment:
that on 1-aixs of 'conv2' -> those on ...

Reply:
Fixed.